### PR TITLE
Fix milestone appliance and permission checks

### LINKED ISSUE #4270: Milestone doesn't get assigned via API
- Gitea version (or commit ref): 85414d8
- Operating system: elementary os / Ubuntu 16.04
- Database (use `[x]`):
  - [x] MySQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes
- Log gist:

## Description

If I try to create an issue via the API and I try to assign a milestone in that request it won't be set.

Request:
```curl
curl -X POST "https://try.gitea.io/api/v1/repos/JonasFranzDEV/gitea/issues" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{  \"body\": \"Test\",  \"milestone\": 165,  \"title\": \"Test Milestone\"}"
```

Response:
```json
{
  "id": 1556,
  "url": "https://try.gitea.io/api/v1/repos/JonasFranzDEV/gitea/issues/50",
  "number": 50,
  "user": {
    "id": 1609,
    "login": "JonasFranzDEV",
    "full_name": "",
    "email": "info@jonasfranz.software",
    "avatar_url": "https://try.gitea.io/avatars/a85562fb670c0233a9bf1f61b2c6c4ee",
    "language": "de-DE",
    "username": "JonasFranzDEV"
  },
  "title": "Test Milestone",
  "body": "Test",
  "labels": [],
  "milestone": null,
  "assignee": null,
  "assignees": null,
  "state": "open",
  "comments": 0,
  "created_at": "2018-06-18T13:08:57Z",
  "updated_at": "2018-06-18T13:08:57Z",
  "closed_at": null,
  "due_date": null,
  "pull_request": null
}
```
`milestone` is `null` but should be set.

### PR DESCRIPTION
**Pull request by jonasfranz**
Fixes #4270 and adds missing permission check which got removed.

Without the permission check every user can assign labels, etc to new issues if created via API.

### CODE REVIEW

**Review by techknowlogick — COMMENTED** [review]
Please add Gitea copyright notice to header of this file

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -178,15 +178,22 @@ func CreateIssue(ctx *context.APIContext, form api.CreateIssueOption) {
...
+			if models.IsErrUserNotExist(err) {
+				ctx.Error(422, "", fmt.Sprintf("Assignee does not exist: [name: %s]", err))
+			} else {
+				ctx.Error(500, "AddAssigneeByName", err)
+			}
+			return
 		}
-		return
+	} else {
+		// setting labels is only allowed if user is writter
```
Comment (lunny) [inline routers/api/v1/repo/issue.go]: comment should be `... is not a writer`.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
Otherwise LGTM

**Comment by lafriks** [comment]
@techknowlogick added copyright
@lunny fixed comment

**Review by techknowlogick — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**